src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -141,10 +152,8 @@ export class IdentityProviderHttpHandler extends HttpHandler {
     let route = this.getRouteMatch(url);
 
     // In case the request targets the IDP entry point the prompt determines where to go
-    if (!route && (url === '/' || url === '')) {
-      const provider = await this.providerFactory.getProvider();
-      const interactionDetails = await provider.interactionDetails(request, response);
-      route = this.getPromptMatch(interactionDetails.prompt.name);
+    if (!route && interaction && (url === '/' || url === '')) {
```

Comment:
I'm a bit confused by the `url` magic here.

First, naming:

```
const pathName = request.url.slice(this.idpPath.length);
```

second, using `ensureTrailingSlash` or the opposite on `this.idpPath` in the constructor, can we simplify the check to `pathName.length === 0`?

Reply:
> second, using `ensureTrailingSlash` or the opposite on `this.idpPath` in the constructor, can we simplify the check to `pathName.length === 0`?

I wanted to support calls to both `/idp` and `/idp/` (although in practice it should always be `/idp/`) so will change it to `ensureTrailingSlash(pathName).length === 0`. Only in the if-statement itself though so it doesn't impact the regex matches before.